cofactr/graph.py

```diff
@@ -1302,3 +1303,91 @@ def create_order(
 
         # Remove `/orders/` to get the ID from the resource path.
         return location[8:]
+
+    def create_get_products_by_ids_job(
+        self,
+        ids: List[str],
+        external: Optional[bool] = True,
+        force_refresh: bool = False,
+        schema: Optional[ProductSchemaName] = None,
+        timeout: Optional[int] = None,
+        owner_id: Optional[str] = None,
+        stale_delta: Optional[str] = None,
+        reference: Optional[str] = None,
+    ) -> List[str]:
+        """Create batch product request job.
+
+        Note: Multiple jobs are created are made if more than 250 IDs are provided.
+
+        Args:
+            ids: Cofactr product IDs to match on.
+            external: Whether to query external sources in order to refresh data if applicable.
+            force_refresh: Whether to force re-ingestion from external sources. Overrides
+                `external`.
+            schema: Response schema.
+            timeout: Time to wait (in seconds) for the server to issue a response.
+            owner_id: Specifies which private data to access.
+            stale_delta: How much time has to pass before data is treated as stale. Use "inf" or
+                "infinite" to indicate data should not be refreshed, no matter how old.
+                Examples: "5h", "1d", "1w", "inf"
+            reference: Arbitrary note to associate with the request.
+
+        Returns:
+            A list with one ID for each job that was created.
+        """
+
+        if not schema:
+            schema = self.default_product_schema
+
+        invariant_query_params = drop_none_values(
+            {
+                "schema": schema.value,
+                "external": bool(external),
+                "force_refresh": force_refresh,
+                "stale_delta": stale_delta,
+            }
+        )
+
+        job_ids = []
+
+        for id_batch in batched(ids, n=_MAX_BATCH_SIZE):
+            res = httpx.post(
+                f"{self.url}/jobs/batch-products-requests/",
+                headers=drop_none_values(
+                    {
+                        "X-CLIENT-ID": self.client_id,
+                        "X-API-KEY": self.api_key,
+                    }
+                ),
+                json={
+                    "batch": [
+                        {
+                            "method": "GET",
+                            "relative_url": (
+                                f'?filtering={filtering}&'
+                                + urlencode(invariant_query_params, doseq=True)
```

Comment:
We can safely omit the use of `doseq`:

```suggestion
                                + urlencode(invariant_query_params)
```